### ISSUE TITLE
Computing MRO

### ISSUE THREAD

**Issue by tristanlatr** [mkdocstrings/griffe#4]
Hi @pawamoy,

I've had time to start working on pydocspec again. 

Here is the code I used to compute the MRO. The algorithm is encapsulated in a generic class such that we can use that with any kind of objects representing a class.

https://github.com/tristanlatr/pydocspec/blob/main/pydocspec/mro.py

The concrete class: https://github.com/tristanlatr/pydocspec/blob/fd25e4eaf47de6fca6706283296316a6ae515955/pydocspec/processor.py#L117

Here are the tests: https://github.com/tristanlatr/pydocspec/blob/main/pydocspec/test/test_mro.py

Hope this helps. 

PS: there is a lot of overlap between griffe and pydocspec.. maybe there is something to do about it?

**Comment by pawamoy** [comment]
Hi @tristanlatr :slightly_smiling_face: 

> I've had time to start working on pydocspec again.

Nice, I already looked a bit what you recently committed (quite of lot of code!), I'll try pydocspec again soon :+1: 
Thanks for the links to your code for computing the MRO, that will be helpful indeed.

> PS: there is a lot of overlap between griffe and pydocspec.. maybe there is something to do about it?

Yeah I was not a really good team player here... doing things on my own... I guess I really needed to scratch that itch.
It's true that there is overlap, but I'm sure there are significant differences as well. And goals.
I did not entirely followed the spec from docspec, but I can totally try to come closer to it (it's not that far already), if it fits the requirements I have for mkdocstrings.

Maybe you had something else in mind?
